packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -0,0 +1,683 @@
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, ["keyCachedSize", "keyCachedTimeout"]);
+		}
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property },
+		value,
+	): Promise<void> => {
+		if (property !== "keyCachedSize" && property !== "keyCachedTimeout") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+		const valueDB = this.endpoint.getNodeUnsafe()!.valueDB;
+
+		let keyCachedSize = value;
+		let keyCachedTimeout: number = value;
+		if (property === "keyCachedSize") {
+			const oldKeyCachedTimeout = valueDB.getValue<number>(
+				getKeyCachedTimeoutStateValueID(this.endpoint.index),
+			);
+			if (oldKeyCachedTimeout == undefined) {
+				throw new ZWaveError(
+					`The "keyCachedSize" property cannot be changed before the key cached timeout is known!`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
+			keyCachedTimeout = oldKeyCachedTimeout;
+		} else {
+			const oldKeyCachedSize = valueDB.getValue<number>(
+				getKeyCachedSizeStateValueID(this.endpoint.index),
+			);
+			if (oldKeyCachedSize == undefined) {
+				throw new ZWaveError(
+					`The "keyCachedTimeout" property cannot be changed before the key cached size is known!`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
+			keyCachedSize = oldKeyCachedSize;
+		}
+		await this.setConfiguration(keyCachedSize, keyCachedTimeout);
+	};
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "dataTypeSupported":
+			case "eventTypeSupported":
+			case "keyCachedSizeMin":
+			case "keyCachedSizeMax":
+			case "keyCachedTimeoutMin":
+			case "keyCachedTimeoutMax":
+				return (await this.getEventSupported())?.[property];
+			case "keySupported":
+				return (await this.getKeySupported())?.[property];
+			case "keyCachedSize":
+			case "keyCachedTimeout":
+				return (await this.getConfiguration())?.[property];
+		}
+		throwUnsupportedProperty(this.ccId, property);
+	};
+}
+
+@commandClass(CommandClasses["Entry Control"])
+@implementedVersion(1)
+export class EntryControlCC extends CommandClass {
+	declare ccCommand: EntryControlCommand;
+
+	public async interview(complete: boolean = true): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Entry Control"].withOptions({
+			priority: MessagePriority.NodeQuery,
+		});
+
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: `${this.constructor.name}: doing a ${
+				complete ? "complete" : "partial"
+			} interview...`,
+			direction: "none",
+		});
+
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "requesting entry control supported keys...",
+			direction: "outbound",
+		});
+
+		{
```

Comment:
Interesting approach with the blocks. I think I still prefer better names for `resp` over this trick :)